server/gorm/client.go

```diff
@@ -301,3 +307,32 @@ func (c *Client) Run(ctx context.Context, q storage.Query) storage.Iterator {
 		return nil
 	}
 }
+
+func (c *Client) GetRecentSpecRevisions(ctx context.Context, offset int32, projectID, apiID, versionID string) storage.Iterator {
+	mylock()
+	defer myunlock()
+
+	op := c.db.Select("specs.*").
+		Table("specs").
+		Joins("JOIN (?) AS grp ON specs.project_id = grp.project_id AND specs.api_id = grp.api_id AND specs.version_id = grp.version_id AND specs.spec_id = grp.spec_id AND specs.revision_create_time = grp.recent_create_time",
```

Comment:
Updated. The query is definitely complicated, but all of its parts are "out of the box" SQL features that can be searched for more info. I added explanations about why it's being done this way since that isn't searchable.